### PR TITLE
Create AWS Create Policy Version to allow all resources

### PR DESCRIPTION
**Pull request by praveenthepro**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: Hunting Queries/AWSCloudTrail/AWS Create Policy Version to allow all resources [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    dataTypes:
+      - AWSCloudTrail
+tactics:
+  - Privilege Escalation
+relevantTechniques:
+  - T1078
+query: |
+
+ AWSCloudTrail
+ | where EventSource == "iam.amazonaws.com" and EventName == "CreatePolicyVersion" and isempty(ErrorMessage) and UserAgent !has "Terraform"
```
Comment (ashwin-patil) [inline Hunting Queries/AWSCloudTrail/AWS Create Policy Version to allow all resources:21]: Lets remove UserAgent filter.. its too broad to filter all terraform events. and add custom allowlist to filter out.

Comment (ashwin-patil) [inline Hunting Queries/AWSCloudTrail/AWS Create Policy Version to allow all resources:21]: Add CreatePolicy to the event list. i tested the query and it works for both events.  
`EventName in ("CreatePolicy","CreatePolicyVersion")`

File: Hunting Queries/AWSCloudTrail/AWS Create Policy Version to allow all resources [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ AWSCloudTrail
+ | where EventSource == "iam.amazonaws.com" and EventName == "CreatePolicyVersion" and isempty(ErrorMessage) and UserAgent !has "Terraform" 
+ | extend PolicyDocument = tostring(parse_json(RequestParameters).policyDocument),UserIdentityUserName = iff(isnotempty(UserIdentityUserName), UserIdentityUserName, tostring(split(UserIdentityArn,'/')[-1]))
+ | where isnotempty(PolicyDocument)
+ | mv-expand todynamic(PolicyDocument)
+ | evaluate bag_unpack(PolicyDocument)
+ | mv-expand todynamic(Statement)
+ | evaluate bag_unpack(Statement)
+ | where isnotempty(Action) and Effect == "Allow" and Resource == "*"
+ | summarize EventCount = count(), StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), Policy_added = make_set(Action) by UserIdentityUserName, EventSource, EventName,UserAgent,AWSRegion,SourceIpAddress
```
Comment (ashwin-patil) [inline Hunting Queries/AWSCloudTrail/AWS Create Policy Version to allow all resources:29]: Add a filter for interesting actions.

`| where Action in (ActionToMonitor)`

**Review by ashwin-patil — CHANGES_REQUESTED** [review]

**Comment by v-dvedak** [comment]
@praveenthepro, Can you please address the comments shared?

**Comment by v-marimanda** [comment]
@praveenthepro  can you please take look  on above comments.

**Comment by praveenthepro** [comment]
Currently this use case is discussion for clubbing this with another privilege escalation use case. Hopefully will update the changes next week.

**Comment by v-spadarthi** [comment]
@praveenthepro : Could you please update if any ?

**Comment by praveenthepro** [comment]
This is an dublicate use case, which is already covered in the below PRhttps://github.com/Azure/Azure-Sentinel/commit/d3683d8deaee64938d434d51a348751362e4f05f

**Pull request closed without merging**